### PR TITLE
docs: analyze memory quota

### PR DESCRIPTION
**Pull request by chrysan**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -396,6 +396,18 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
     - `tidb_auto_analyze_start_time='01:00 +0000'`
     - `tidb_auto_analyze_end_time='03:00 +0000'`
 
+### `tidb_mem_quota_analyze`
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`0`
+- 单位：字节
+- 这个变量用来控制 TiDB 更新统计信息时总的内存占用，包括用户执行的 [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) 和 TiDB 后台自动执行的统计信息更新任务。当总的内存占用超过这个阈值时，用户执行的 ANALYZE 会被终止退出，并通过错误信息提示用户尝试更小的采样率或稍后重试。TiDB 后台自动执行的统计信息更新任务会使用默认采样率重试一次。当变量设置为负数或零时，不对 ANALYZE 的内存限制做控制。
```
Comment (xuyifangreeneyes) [inline system-variables.md:405]: 不对 ANALYZE 进行内存限制 听起来更通顺一些？

Comment (xuyifangreeneyes) [inline system-variables.md:405]: AUTO ANALYZE 是检查采样率是否大于默认采样率，如果是才重试。

File: statistics.md [hunk 2]
```diff
@@ -427,6 +427,17 @@ SHOW ANALYZE STATUS [ShowLikeOrWhere];
 | start_time | 任务开始执行的时间 |
 | state | 任务状态，包括 pending（等待）、running（正在执行）、finished（执行成功）和 failed（执行失败）|
 
+当设置了 `tidb_mem_quota_analyze` 且 TiDB 后台的统计信息自动更新任务触发了这个阈值时，自动更新任务会重试，失败的任务和重试的任务都可以在 `SHOW ANALYZE STATUS` 的结果中查看。
```
Comment (xuyifangreeneyes) [inline statistics.md:430]: 超过了这个阈值 可能更贴切？

**Review by xuyifangreeneyes — COMMENTED** [review]
rest LGTM

File: system-variables.md (hunk 1 above)
Comment (chrysan) [reply system-variables.md]: Done

File: statistics.md [hunk 3]
```diff
@@ -427,6 +427,17 @@ SHOW ANALYZE STATUS [ShowLikeOrWhere];
 | start_time | 任务开始执行的时间 |
 | state | 任务状态，包括 pending（等待）、running（正在执行）、finished（执行成功）和 failed（执行失败）|
 
+当设置了 [`tidb_mem_quota_analyze`](/system-variables.md#tidb_mem_quota_analyze从-v60-版本开始引入) 且 TiDB 后台的统计信息自动更新任务超过了这个阈值时，自动更新任务会重试。失败的任务和重试的任务都可以在 `SHOW ANALYZE STATUS` 的结果中查看。
```
Comment (Oreoxmt) [inline statistics.md:430]:
```suggestion
当设置了 [`tidb_mem_quota_analyze`](/system-variables.md#tidb_mem_quota_analyze从-v610-版本开始引入) 且 TiDB 后台的统计信息自动更新任务超过了这个阈值时，自动更新任务会重试。失败的任务和重试的任务都可以在 `SHOW ANALYZE STATUS` 的结果中查看。
```

File: system-variables.md [hunk 4]
```diff
@@ -396,6 +396,18 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
     - `tidb_auto_analyze_start_time='01:00 +0000'`
     - `tidb_auto_analyze_end_time='03:00 +0000'`
 
+### `tidb_mem_quota_analyze` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`0`
+- 单位：字节
+- 这个变量用来控制 TiDB 更新统计信息时总的内存占用，包括用户执行的 [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) 和 TiDB 后台自动执行的统计信息更新任务。当总的内存占用超过这个阈值时，用户执行的 `ANALYZE` 会被终止退出，并通过错误信息提示用户尝试更小的采样率或稍后重试。TiDB 后台自动执行的统计信息更新任务若因超内存退出，且使用的采样率高于默认值，会使用默认采样率重试一次。当变量设置为负数或零时，不对 `ANALYZE` 进行内存限制。
```
Comment (Oreoxmt) [inline system-variables.md:405]:
```suggestion
- 这个变量用来控制 TiDB 更新统计信息时的最高总内存占用，包括用户执行的 [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) 和 TiDB 后台自动执行的统计信息更新任务。当总的内存占用超过这个阈值时，用户执行的 `ANALYZE` 会被终止退出，并通过错误信息提示用户尝试更小的采样率或稍后重试。如果 TiDB 后台自动执行的统计信息更新任务因内存超限而退出，且使用的采样率高于默认值，则会使用默认采样率重试一次。当变量设置为负数或零时，不对 `ANALYZE` 进行内存限制。
```

Comment (Oreoxmt) [inline system-variables.md:405]:
```suggestion
- 这个变量用来控制 TiDB 更新统计信息时总的内存占用，包括用户执行的 [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) 和 TiDB 后台自动执行的统计信息更新任务。当总的内存占用超过这个阈值时，用户执行的 `ANALYZE` 会被终止退出，并通过错误信息提示用户尝试更小的采样率或稍后重试。TiDB 后台自动执行的统计信息更新任务若因超内存退出，且使用的采样率高于默认值，会使用默认采样率重试一次。当变量设置为负数或零时，不对更新统计信息进行内存限制。
```

@chrysan 在前面的说明中“更新统计信息”包含“用户执行 `ANALYZE TABLE`” 和 “TiDB 后台自动执行的更新” 两部分，那么当变量设置为负数或零时，是不对用户执行的 `ANALYZE TABLE` 进行内存限制，还是不对用户执行的 `ANALYZE TABLE` 以及 TiDB 后台自动执行的更新进行内存限制呢？

Comment (chrysan) [reply system-variables.md:405]: 包括用户执行的和后台自动执行的

File: system-variables.md [hunk 5]
```diff
@@ -396,6 +396,18 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
     - `tidb_auto_analyze_start_time='01:00 +0000'`
     - `tidb_auto_analyze_end_time='03:00 +0000'`
 
+### `tidb_mem_quota_analyze` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`0`
+- 单位：字节
+- 这个变量用来控制 TiDB 更新统计信息时总的内存占用，包括用户执行的 [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) 和 TiDB 后台自动执行的统计信息更新任务。当总的内存占用超过这个阈值时，用户执行的 `ANALYZE` 会被终止退出，并通过错误信息提示用户尝试更小的采样率或稍后重试。TiDB 后台自动执行的统计信息更新任务若因超内存退出，且使用的采样率高于默认值，会使用默认采样率重试一次。当变量设置为负数或零时，不对更新统计信息进行内存限制。
```
Comment (Oreoxmt) [inline system-variables.md:405]:
```suggestion
- 这个变量用来控制 TiDB 更新统计信息时的最高总内存占用，包括用户执行的 [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) 和 TiDB 后台自动执行的统计信息更新任务。当总的内存占用超过这个阈值时，用户执行的 `ANALYZE` 会被终止退出，并通过错误信息提示用户尝试更小的采样率或稍后重试。如果 TiDB 后台自动执行的统计信息更新任务因内存超限而退出，且使用的采样率高于默认值，则会使用默认采样率重试一次。当变量设置为负数或零时，不对更新统计信息进行内存限制。
```

File: statistics.md [hunk 6]
```diff
@@ -427,6 +427,17 @@ SHOW ANALYZE STATUS [ShowLikeOrWhere];
 | start_time | 任务开始执行的时间 |
 | state | 任务状态，包括 pending（等待）、running（正在执行）、finished（执行成功）和 failed（执行失败）|
 
+当设置了 [`tidb_mem_quota_analyze`](/system-variables.md#tidb_mem_quota_analyze-从-v610-版本开始引入) 且 TiDB 后台的统计信息自动更新任务超过了这个阈值时，自动更新任务会重试。失败的任务和重试的任务都可以在 `SHOW ANALYZE STATUS` 的结果中查看。
```
Comment (xuyifangreeneyes) [inline statistics.md:430]: TiDB 后台的统计信息自动更新任务内存占用超过了这个阈值时

**Review by xuyifangreeneyes — APPROVED** [review]
rest LGTM

File: system-variables.md [hunk 7]
```diff
@@ -396,6 +396,18 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
     - `tidb_auto_analyze_start_time='01:00 +0000'`
     - `tidb_auto_analyze_end_time='03:00 +0000'`
 
+### `tidb_mem_quota_analyze` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`0`
+- 单位：字节
+- 这个变量用来控制 TiDB 更新统计信息时的最高总内存占用，包括用户执行的 [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) 和 TiDB 后台自动执行的统计信息更新任务。当总的内存占用超过这个阈值时，用户执行的 `ANALYZE` 会被终止退出，并通过错误信息提示用户尝试更小的采样率或稍后重试。如果 TiDB 后台自动执行的统计信息更新任务因内存超限而退出，且使用的采样率高于默认值，则会使用默认采样率重试一次。当变量设置为负数或零时，不对更新统计信息进行内存限制。
```
Comment (chrysan) [inline system-variables.md:405]:
```suggestion
- 这个变量用来控制 TiDB 更新统计信息时的最高总内存占用，包括用户执行的 [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) 和 TiDB 后台自动执行的统计信息更新任务。当总的内存占用超过这个阈值时，用户执行的 `ANALYZE` 会被终止退出，并通过错误信息提示用户尝试更小的采样率或稍后重试。如果 TiDB 后台自动执行的统计信息更新任务因内存超限而退出，且使用的采样率高于默认值，则会使用默认采样率重试一次。当变量设置为负数或零时，不对更新统计信息的前后台任务进行内存限制。
```

File: statistics.md (hunk 6 above)
Comment (chrysan) [reply statistics.md]: Done

File: system-variables.md [hunk 8]
```diff
@@ -396,6 +396,18 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
     - `tidb_auto_analyze_start_time='01:00 +0000'`
     - `tidb_auto_analyze_end_time='03:00 +0000'`
 
+### `tidb_mem_quota_analyze` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`0`
+- 单位：字节
+- 这个变量用来控制 TiDB 更新统计信息时的最高总内存占用，包括用户执行的 [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) 和 TiDB 后台自动执行的统计信息更新任务。当总的内存占用超过这个阈值时，用户执行的 `ANALYZE` 会被终止退出，并通过错误信息提示用户尝试更小的采样率或稍后重试。如果 TiDB 后台自动执行的统计信息更新任务因内存超限而退出，且使用的采样率高于默认值，则会使用默认采样率重试一次。当变量设置为负数或零时，不对更新统计信息的前后台任务进行内存限制。
```
Comment (TomShawn) [inline system-variables.md:405]:
```suggestion
- 这个变量用来控制 TiDB 更新统计信息时的最大总内存占用，包括用户执行的 [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) 和 TiDB 后台自动执行的统计信息更新任务。当总的内存占用超过这个阈值时，用户执行的 `ANALYZE` 会被终止退出，并通过错误信息提示用户尝试更小的采样率或稍后重试。如果 TiDB 后台自动执行的统计信息更新任务因内存超限而退出，且使用的采样率高于默认值，则会使用默认采样率重试一次。当该变量值为负数或零时，TiDB 不对更新统计信息的前后台任务进行内存限制。
```

File: system-variables.md [hunk 9]
```diff
@@ -396,6 +396,18 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+
+> **注意：**
+>
+> 只有在 TiDB 的启动配置文件中开启了 `run-auto-analyze` 选项，该 TiDB 才会触发 `auto_analyze`。
```
Comment (TomShawn) [inline system-variables.md:409]: 不太确定是否为「实例」，但需要一个单位
```suggestion
> 只有在 TiDB 的启动配置文件中开启了 `run-auto-analyze` 选项，该 TiDB 实例才会触发 `auto_analyze`。
```

**Review by TomShawn — APPROVED** [review]
rest LGTM

File: system-variables.md (hunk 9 above)
Comment (chrysan) [reply system-variables.md:409]: 不是实例，一个 TiDB 集群只有一个实例跑 auto_analyze

File: system-variables.md [hunk 10]
```diff
@@ -396,6 +396,18 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
     - `tidb_auto_analyze_start_time='01:00 +0000'`
     - `tidb_auto_analyze_end_time='03:00 +0000'`
 
+### `tidb_mem_quota_analyze` <span class="version-mark">从 v6.1.0 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:399]: 变量尽量按照字母顺序排列

**Pull request merged**